docs/DesignDocs/CryptoLibConfiguration.md

```diff
@@ -0,0 +1,115 @@
+# Configuration Parity Analysis between MbedTLS and OpenSSL Crypto Libraries
+
+This document compares the build-time configuration of MbedTLS, which
+has been extenstively reviewed when the library was ported to OE,
+against that of OpenSSL, which OE recently starts to support. The goal of this
+analysis is to ensure the parity between the two. For certain configurations
+that cannot achieve parity, the analysis also tries to reason out the gap.
+
+*Note:* The following table only lists the options that are revelant to
+the cryptographic algorithms. The table uses *CONFIG* or *!CONFIG* to indicate the
+corresponding configuration is set or not.
+
+## Parity analysis based on MbedlTLS
```

Comment:
Great table, really helpful for doing the parity analysis! I would suggest a couple of changes to make it easier to make sense of:
- Break apart the option name from the configuration of that option OE uses to make it easier to parse because a `!` in mbedTLS column reflects a configuration state, and the `!` in the OpenSSL column reflects the parity setting which makes following what the comments refer to as "supported" unclear.
- Clarify which settings are OE specific-choices vs. mbedTLS defaults. We generally don't care so much about the latter cases being different.  
- Remove the OE comments about why some options are configured from the Comment column, and use that column just to talk about how we achieve parity behavior in OpenSSL, or what the resulting gap is if not. 
   - Alternatively, move the comments about the OE decisions to the OE configuration column so that it's clear what the comment specifically applies to.
- Highlight the rows where we have explicit mismatches in behavior for configurations OE has chosen, or are of concern. e.g. add a bold phrase that is easy to search like **No parity**. 

For example, consider something like:

MbedTLS option | OE configuration | Equivalent OpenSSL option | Comment
:---|:---|:---|:---|
MBEDTLS_CIPHER_MODE_CFB | Disabled (by OE as unsafe block cipher mode) | N/A | **No parity.** OpenSSL always supports CFB mode and cannot be disabled by OE. |
MBEDTLS_RSA_NO_CRT | Disabled (MbedTLS default) | N/A | Default OpenSSL behavior is to use Chinese Remainder Theorem in RSA, so while OpenSSL does not support disabling its use, it has parity behavior in OE with mbedTLS. |
MBEDTLS_KEY_EXCHANGE_PSK_ENABLED | Disabled (by OE as not providing perfect forward secrecy and not recommended for future use.) | OPENSSL_NO_PSK | OE will use the `no-psk` configuration to disable PSK in OpenSSL for parity. |
MBEDTLS_MD5_C | Enabled (MbedTLS default, OE doesn't override this because it is still commonly used for backwards compatibility) | !OPENSSL_NO_MD5 | OpenSSL supports MD5 by default, so we keep that behavior for parity. |